app/controllers/api_controller/providers.rb

```diff
@@ -138,7 +138,7 @@ def fetch_provider_data(provider_klass, data, options = {})
       raise BadRequestError, "Invalid Provider attributes #{invalid_keys.join(', ')} specified" if invalid_keys.present?
 
       specify_zone(provider_data, data, options)
-      provider_data
+      provider_data.deep_symbolize_keys
```

Comment:
> Why was this change needed? Do any of the tests fail without it?

@imtayadeway the `connection_configurations=` method looks for symbols [L267](https://github.com/ManageIQ/manageiq/blob/master/app/models/ext_management_system.rb#L267) and [L276](https://github.com/ManageIQ/manageiq/blob/master/app/models/ext_management_system.rb#L276) in each connection option. If we do not convert the connection options keys to symboles before this method gets them we will get an error:
`"Could not create the new provider - undefined method key?' for nil:NilClass",`

We can also do the conversion on the `connection_configurations=` side ... WDYT ?